### ISSUE TITLE
Support for compilerOptions.types

### ISSUE THREAD

**Issue by jupl** [TypeStrong/typedoc#311]
Given this `tsconfig.json`:

``` json
{
  "compilerOptions": {
    "jsx": "react",
    "moduleResolution": "node",
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "sourceMap": true,
    "strictNullChecks": true,
    "target": "es2015",
    "types": [
      "jest",
      "node"
    ]
  }
}
```

TypeDoc fails to run with the following error:

```
Error: Unknown option `types.0`.
Error: Unknown option `types.1`.
```

**Comment by aciccarello** [comment]
This is a remaining issue from #234. A PR for this would be great. Also related is #305 (supporting typeroots)

**Comment by Radagaisus** [comment]
A quick patch is to add `typeRoots` and `types` to the [ignored compiler options](https://github.com/TypeStrong/typedoc/blob/22ccb660ae6eb5b3e136dc277ecc1d62fb04d016/src/lib/utils/options/sources/typescript.ts#L16-L20). Though, for better support of future TypeScript versions, and an easier upgrade path, it’s best to switch to a whitelist.

**Comment by aciccarello** [comment]
Does anyone know if this was fixed in #359?

**Comment by jupl** [comment]
@aciccarello The specific error does not show as of the latest version in NPM but errors are being thrown when trying to compile docs such that it appears that type `types` option is being completely ignored by Typedoc.

**Comment by jiayihu** [comment]
I can confirm that `types` option is ignored and I can't compile docs

**Comment by Gerrit0** [comment]
This appears to have been fixed with #449
